Content.Shared/Teleportation/Systems/SwapTeleporterSystem.cs

```diff
@@ -167,11 +181,48 @@ public void DoTeleport(Entity<SwapTeleporterComponent, TransformComponent> ent)
             return;
         }
 
+
+        // Prevents teleporting to the polymorph zone or the cryosleep zone.
+        // Don't unlink because it might be temporary (polymorph). Let them figure it out and unlink it themselves.
+        if (_map.IsPaused(Transform(teleEnt).MapID) || _map.IsPaused(Transform(otherTeleEnt).MapID))
+        {
+            _popup.PopupEntity(Loc.GetString("swap-teleporter-popup-teleport-fail",
+                ("entity", Identity.Entity(linkedEnt, EntityManager))),
+                teleEnt,
+                teleEnt,
+                PopupType.MediumCaution);
+            return;
+        }
+
         _popup.PopupClient(Loc.GetString("swap-teleporter-popup-teleport-other",
             ("entity", Identity.Entity(linkedEnt, EntityManager))),
             teleEnt,
             otherTeleEnt,
             PopupType.MediumCaution);
+
+        // break pulls before teleport so we dont break shit
+        if (TryComp<PullableComponent>(teleEnt, out var pullable) && pullable.BeingPulled)
+        {
+            _pulling.TryStopPull(teleEnt, pullable);
+        }
+
+        if (TryComp<PullerComponent>(teleEnt, out var pullerComp)
+            && TryComp<PullableComponent>(pullerComp.Pulling, out var subjectPulling))
+        {
+            _pulling.TryStopPull(pullerComp.Pulling.Value, subjectPulling);
+        }
+        // both sides
+        if (TryComp<PullableComponent>(otherTeleEnt, out var otherPullable) && otherPullable.BeingPulled)
+        {
+            _pulling.TryStopPull(otherTeleEnt, otherPullable);
+        }
+
+        if (TryComp<PullerComponent>(otherTeleEnt, out var otherPullerComp)
+            && TryComp<PullableComponent>(otherPullerComp.Pulling, out var otherSubjectPulling))
+        {
+            _pulling.TryStopPull(otherPullerComp.Pulling.Value, otherSubjectPulling);
+        }
```

Comment:
i feel like this is better served by making pulling logic more robust so it actually breaks joints when outside of a reasonable distance rather than just keeping it specifically here.

Reply:
Seems to be outside of the scope of this PR, but I've raised a separate issue bringing this up. #31214